src/main/java/com/frameworkium/core/ui/listeners/VideoListener.java

```diff
@@ -0,0 +1,24 @@
+package com.frameworkium.core.ui.listeners;
+
+import com.frameworkium.core.ui.capture.VideoCapture;
+import org.openqa.selenium.remote.SessionId;
+import org.testng.ITestResult;
+import org.testng.TestListenerAdapter;
+
+import static com.frameworkium.core.ui.capture.VideoCapture.isRequired;
+import static com.frameworkium.core.ui.tests.BaseTest.getDriver;
+import static com.frameworkium.core.ui.tests.BaseTest.getFrameworkDriver;
+
+public class VideoListener extends TestListenerAdapter {
+
+    private VideoCapture videoCapture = new VideoCapture();

```

Comment:
I'm not sure how the threading works with Listeners, so to prevent any issues and to keep the method simple I'd suggest to remove this and in-line `new VideoCapture()`.